lib/pliny/helpers/params.rb

```diff
@@ -21,10 +21,21 @@ def parse_body_params
     def load_params(data)
       # Sinatra 1.x only supports the method. Sinatra 2.x only supports the class
       if defined?(Sinatra::IndifferentHash)
-        Sinatra::IndifferentHash[data]
+        indifferent_params_v2(data)
       else
         indifferent_params(data)
       end
     end
+
+    def indifferent_params_v2(data)
+      case data
+      when Hash
+        Sinatra::IndifferentHash[data.to_hash]
+      when Array
+        data.to_ary.map { |item| Sinatra::IndifferentHash[item] }
```

Comment:
@stevenharman I think the `to_ary` is redundant here because have verified it is an `Array`.  Could you also make this recursive to support arrays of arrays plus arrays of primitives, etc.

Reply:
Done.